pkg/container-hook/tracer.go

```diff
@@ -537,44 +550,41 @@ func (n *ContainerNotifier) monitorRuntimeInstance(bundleDir string, pidFile str
 		// podman might install config.json in the userdata directory
 		configJSONPath = filepath.Join(bundleDir, "userdata", "config.json")
 		if _, err := os.Stat(configJSONPath); errors.Is(err, os.ErrNotExist) {
-			pidFileDirNotify.File.Close()
 			return fmt.Errorf("config not found at %s", configJSONPath)
 		}
 	}
-	err = pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, configJSONPath)
+	err = n.pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, configJSONPath)
 	if err != nil {
-		pidFileDirNotify.File.Close()
 		return fmt.Errorf("marking %s: %w", configJSONPath, err)
 	}
 
-	// similar to config.json, we ignore passwd file if it exists
-	passwdPath := filepath.Join(bundleDir, "passwd")
-	if _, err := os.Stat(passwdPath); !errors.Is(err, os.ErrNotExist) {
-		err = pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, passwdPath)
+	// To reduce noise, ignore additional files (best effort)
+	ignoreFileList := []string{
+		"passwd",
+		"log.json",
+		"runtime",
+	}
+	for _, ignoreFile := range ignoreFileList {
+		ignoreFilePath := filepath.Join(bundleDir, ignoreFile)
+		err := n.pidFileDirNotify.Mark(unix.FAN_MARK_ADD|unix.FAN_MARK_IGNORED_MASK, unix.FAN_ACCESS_PERM, unix.AT_FDCWD, ignoreFilePath)
```

Comment:
I removed the test with `os.Stat()` introduces the possibility of TOCTOU issues: if containerd deletes the container directory completely between the `os.Stat` and the `Mark`, that would generate an error which should be ignored.

This mark with `FAN_MARK_IGNORED_MASK` is best effort to reduce the noise. Some container runtimes don't use those files.

The check on passwd was added in #1742 to support cri-o; this was necessary because `checkFilesAreIdentical` returned an error when one of the file didn't exist. In this PR, the error from `checkFilesAreIdentical` is no longer fatal.

Also, we should not rely on `FAN_MARK_IGNORED_MASK` working properly because Linux < 5.9 (https://github.com/torvalds/linux/commit/497b0c5a7c0688c1b100a9c2e267337f677c198e) didn't implement `FAN_MARK_IGNORED_MASK` on files when the parent directory is the object being watched.